lib/commands/command-plugin-update.bash

```diff
@@ -30,11 +30,21 @@ update_plugin() {
   plugin_remote_default_branch=$(git --git-dir "$plugin_path/.git" --work-tree "$plugin_path" ls-remote --symref origin HEAD | awk '{ sub(/refs\/heads\//, ""); print $2; exit }')
   local gitref=${3:-${plugin_remote_default_branch}}
   logfile=$(mktemp)
-  {
-    printf "Updating %s to %s\\n" "$plugin_name" "$gitref"
-    git --git-dir "$plugin_path/.git" --work-tree "$plugin_path" fetch --prune --update-head-ok origin "$gitref:$gitref"
-    git --git-dir "$plugin_path/.git" --work-tree "$plugin_path" -c advice.detachedHead=false checkout --force "$gitref"
-  } >"$logfile" 2>&1
+
+  local is_detached=false
+  [ $(git --git-dir "$plugin_path/.git" rev-parse --abbrev-ref --symbolic-full-name HEAD) = "HEAD" ] && is_detached="true"
+  if [[ "$is_detached" == "true" ]]; then
```

Comment:
FWIW this would probably be much simpler but I chose a more verbose copy to make the command more clear. Or perhaps a simple comment would serve this better

```suggestion
  if [ $(git --git-dir "$plugin_path/.git" rev-parse --abbrev-ref --symbolic-full-name HEAD) = "HEAD" ]; then
```